packages/git/src/node/dugite-git.ts

```diff
@@ -0,0 +1,210 @@
+/*
+* Copyright (C) 2017 TypeFox and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+import * as Path from 'path';
+import { Git, GitUtils } from '../common/git';
+import { git } from 'dugite-extra/lib/core/git';
+import { injectable, inject } from "inversify";
+import { FileUri } from '@theia/core/lib/node/file-uri';
+import { getStatus } from 'dugite-extra/lib/command/status';
+import { clone } from 'dugite-extra/lib/command/clone';
+import { createCommit } from 'dugite-extra/lib/command/commit';
+import { getTextContents, getBlobContents } from 'dugite-extra/lib/command/show';
+import { createBranch, deleteBranch, renameBranch, listBranch } from 'dugite-extra/lib/command/branch';
+import { stage, unstage } from 'dugite-extra/lib/command/stage';
+import { locateRepositories } from './git-repository-locator';
+import { WorkspaceServer } from '@theia/workspace/lib/common/workspace-protocol';
+import { Repository, WorkingDirectoryStatus, FileChange, FileStatus } from '../common/model';
+import { IStatusResult, IAheadBehind, AppFileStatus, WorkingDirectoryStatus as DugiteStatus, FileChange as DugiteFileChange } from 'dugite-extra/lib/model/status';
+
+/**
+ * `dugite-extra` based Git implementation.
+ */
+@injectable()
+export class DugiteGit implements Git {
+
+    constructor(
+        @inject(WorkspaceServer) private readonly workspace: WorkspaceServer) {
```

Comment:
it would be nice to prefer `protected` over `private` to allow subclasses to override and access the state.

Reply:
If you prefer that, we use protected then.

Reply:
it is just a pain, for example, to work around some bugs in JDT because everything is private.